FIO-8375: allow date time to have manual input

ZenMasterJacob20011 opened the pull request:
## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-8375

## Description

**What changed?**

Previously when manual input was given to the date/time component the input mask would fill to create a complete time on blur. Now when the input triggers a blur event the manual input will not fill the input mask but instead be left incomplete. I have also made changes so that the validation will trigger as you are typing a manual input.

**Why have you chosen this solution?**

This solution allows you to use both the flatpickr and manual input without needing to switch one or the other off

## Breaking Changes / Backwards Compatibility

This PR should be tested extensively for correct validation, and correct submission data when submitted. To allow for date time input mask to remain incomplete upon clicking outside of the input required a lot of fighting with the Flatpickrs event listeners. 

## Dependencies

N/A

## How has this PR been tested?

Manually tested and automated test

## Checklist:

- [x] I have completed the above PR template
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings
- [x] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [x] New and existing unit/integration tests pass locally with my changes
- [x] Any dependent changes have corresponding PRs that are listed above

TanyaGashtold commented on src/widgets/CalendarWidget.js:
```diff
@@ -142,9 +135,13 @@ export default class CalendarWidget extends InputWidget {
+        setTimeout(()=> {
+          this.calendar._input.value = this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value;
+          this._input.value = this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value
+          this.emit('update');
```
1)  this.calendar._input.value  and this._input.value has the same value, You can make this record shorter:
this.calendar._input.value = this._input.value =  this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value;
2) It is not a good practice to use setTimeout often. We need to try to avoid them where it is possible. If this code must be executed after flatpickr sets the value, can we use it inside some other flatpickr hook, like onValueUpdate or onChange (https://flatpickr.js.org/events/)?

TanyaGashtold commented on src/widgets/CalendarWidget.js:
```diff
@@ -424,20 +421,24 @@ export default class CalendarWidget extends InputWidget {
-      }
+      // Need to give time for the input mask to update so that the manualInputValue can be properly set
+      setTimeout(() => {
+        if (this.settings.allowInput && this.settings.currentValue !== event.target.value) {
```
Can we try to avoid setTimeout here as well?

ZenMasterJacob20011 commented:
@TanyaGashtold I didn't realize deleting the comments in my ide would also delete the comments in github as well. I have removed the setTimeouts as requested. I also added some inputEvents for the tests because they would not pass without them. I believe that it is necessary to trigger these input events because manual values require input events to occur for them to work properly
